### PR TITLE
Remove deprecated classnames from Navigation Link block

### LINKED ISSUE #35237: Remove deprecated classnames from Navigation Link block
## What problem does this address?

After #33918 and #34171, there are a some extra classnames on Navigation Link block that are no longer needed. It would be good to remove them before the block lands in core.

List of classnames we can remove: 

* `wp-block-navigation-link__container`
* `wp-block-navigation-link__content` (one of the e2e tests is using this one as a selector so we should also update that)
* `wp-block-navigation-link__submenu-icon`

### PR DESCRIPTION
**Pull request by carolinan**
## Description

Fixes https://github.com/WordPress/gutenberg/issues/35237, #35238

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by talldan** [comment]
There are some separate styles for the navigation editor that depend on those classnames which would also need to be updated:
https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-navigation/src/components/editor/style.scss

File: packages/block-library/src/navigation-submenu/edit.js [hunk 1]
```diff
@@ -576,7 +576,7 @@ export default function NavigationSubmenuEdit( {
 						<RichText
 							ref={ ref }
 							identifier="label"
-							className="wp-block-navigation-link__label"
+							className="wp-block-navigation-litem__label"
```
Comment (tellthemachines) [inline packages/block-library/src/navigation-submenu/edit.js]: Typo: should be "item" not "litem".

File: packages/components/src/navigation/README.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ The structural CSS for the navigation block targets generic classnames across me
 - `.wp-block-navigation__submenu-container` is applied to submenus to main menu items.
```
Comment (tellthemachines) [inline packages/components/src/navigation/README.md:51]: Not sure what these docs about Nav block classnames are doing in the components package 🤔 they should be in the block folder instead. No need to fix that here though, I've created #35445 to address it separately.

**Review by tellthemachines — APPROVED** [review]
Thanks for fixing this! Code is looking good aside from a tiny typo I commented on below, so I'm going to go ahead and approve the PR anyway.

File: packages/block-library/src/navigation-submenu/edit.js (hunk 1 above)
Comment (carolinan) [inline packages/block-library/src/navigation-submenu/edit.js]:
```suggestion
							className="wp-block-navigation-item__label"
```

**Pull request merged**